device/tt_cluster_descriptor_types.h

```diff
@@ -4,24 +4,63 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#pragma once 
+#pragma once
 
-#include <tuple>
+#include <fmt/format.h>
 
-using chip_id_t = int;
-using ethernet_channel_t = int;
-using eth_coord_t = std::tuple<int, int, int, int>;  // x, y, rack, shelf
+#include <functional>
+#include <type_traits>
+
+
+namespace tt::umd {
+
+enum class chip_id : int {
```

Comment:
Having this as an enum can be a bit confusing, since you're using it such that any int value can be assigned to it, and it doesn't necessarily have a corresponding enum constant. However, I do see the value in using an explicit name for a type to have a clearer requirement  and expectations over what is expected from an interface.

the eth_coord clearly benefits in readability from being defined as a struct, but whats the reasoning behind changing using chip_id_t to enum class?
You would then have define a constexpr int c_invalid_chip_id = -1;

Reply:
Ah actually, I saw the attached tt-metal issue, this ensures explicit conversion. And enum class is a different thing than enum :) LGTM